File: src/main/java/io/github/dqualizer/dqlang/types/adapter/request/Checks.java
package io.github.dqualizer.dqlang.types.adapter.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

import java.util.Set;

@Getter
@ToString
@AllArgsConstructor
@NoArgsConstructor
public class Checks {

    @JsonProperty("status_codes")
    private Set<Integer> statusCodes;
    @JsonProperty("duration")
    private int duration;
}

File: src/main/kotlin/io/github/dqualizer/dqlang/types/adapter/k6/request/Checks.kt
package io.github.dqualizer.dqlang.types.adapter.k6.request

import com.fasterxml.jackson.annotation.JsonProperty

data class Checks(
  @JsonProperty("status_codes")
  val statusCodes: Set<Int>? = null,

  val duration: Int? = 0
)

